wire-gradle-plugin/src/main/kotlin/com/squareup/wire/gradle/WireExtension.kt

```diff
@@ -37,20 +37,30 @@ open class WireExtension(
   internal val protoPaths = mutableSetOf<String>()
   internal val sourceTrees = mutableSetOf<SourceDirectorySet>()
   internal val protoTrees = mutableSetOf<SourceDirectorySet>()
+  internal val roots = mutableSetOf<String>()
+  internal val prunes = mutableSetOf<String>()
+
+  @Input
+  @Optional
+  fun roots() = roots.toSet()
 
   /**
    * See [com.squareup.wire.schema.WireRun.treeShakingRoots]
    */
-  @get:Input
-  @get:Optional
-  var roots: Array<String>? = null
+  fun root(vararg roots: String) {
+    this.roots.addAll(Arrays.asList(*roots))
```

Comment:
You shouldn't need to create a list. There's an overload for `addAll` that takes an array: https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/add-all.html
```suggestion
    this.roots.addAll(roots)
```

Reply:
good catch.  that was copy-pasta from another method.

Reply:
actually, i'll address all the methods that do this in a follow-up PR.